contracts/eosio.system/src/voting.cpp

```diff
@@ -140,6 +143,10 @@ namespace eosiosystem {
       if( set_proposed_producers( producers ) >= 0 ) {
          _gstate.last_producer_schedule_size = static_cast<decltype(_gstate.last_producer_schedule_size)>( top_producers.size() );
       }
+
+      if( is_savanna_consensus() ) {
+         check(set_next_finalizers_set( top_producer_names ), "set_next_finalizers_set failed (less than 15 producers have registered keys");
```

Comment:
We shouldn't fail in an `onblock`. If we cannot meet the require 21 (not 15) producers, then we should simply avoid changing the proposer policy or finalizer policy.